Danger/Dangerfile.swift

```diff
@@ -1,482 +1,17 @@
-                return .acceptable
-            }
-        }
-        
-        if isHeadBranch(.ci) {
-            // If the PR is created from the branch created by CI, do the CI auto PR modification check.
-            checkCIAutoPRModification(into: &result)
-            
-        } else if isHeadBranch(.version) {
-            // If the PR is created from a version branch, do the release modification check.
-            checkReleaseModification(into: &result)
-            
-        } else {
-            // Otherwise, do the develop modification check.
-            checkDevelopmentModification(into: &result)
-        }
-        
-        return result
-        
-    }
-    
-    func doReleasePRCheck() -> CheckResult {
-        
-        var result = CheckResult(title: "Release PR Check")
-        
-        // Release PR should be created from develop branch.
-        let doHeadBranchCheckTitle = "PR Head Branch Check"
-        result.check(doHeadBranchCheckTitle) {
-            if isHeadBranch(.develop) {
-                return .good
-                
-            } else {
-                fail("Please create a release PR from develop branch.")
-                return .rejected
-            }
-        }
-
-        // Develop PR should be created into main branch
-        let doBaseBranchCheckTitle = "PR Base Branch Check"
-        result.check(doBaseBranchCheckTitle) {
-            if isBaseBranch(.main) {
-                return .good
-                
-            } else {
-                fail("Please create a release PR into main branch.")
-                return .rejected
-            }
-        }
-        
-        // Do the release modification check.
-        checkReleaseModification(into: &result)
-        
-        return result
-        
-    }
-    
-    enum RoutineError: Error {
-        case failedToFindCheckRoutine
-    }
-    
-    func checkPR() throws -> CheckResult {
-        
-        if isDevelopPR {
-            return doDevelopPRCheck()
-            
-        } else if isReleasePR {
-            return doReleasePRCheck()
-            
-        } else {
-            throw RoutineError.failedToFindCheckRoutine
-        }
-        
-    }
-    
-}
-
-// MARK: - Other convenient extensions
-
-private extension Git {
-    
-    var diffFiles: [File] {
-        createdFiles + deletedFiles + modifiedFiles
-    }
-    
-}
-
-private extension String {
-    
-    func substring <S: StringProtocol> (of string: S, options: CompareOptions) -> String? {
-        
-        guard let range = range(of: string, options: options) else {
-            return nil
-        }
-        
-        return String(self[range])
-        
-    }
-    
-}
 
+// MARK: - 便利な拡張
 private extension ProcessInfo {
     
     static var xcTestResultPath: String {
         
         // If running on Xcode Cloud it should be able to get path from `$CI_RESULT_BUNDLE_PATH` variable
         processInfo.environment["CI_RESULT_BUNDLE_PATH"] ??
```

Comment:
<!--
  0 failure: 
  1 warning:  Assigning default...
  
  
  DangerID: danger-id-Danger;
  File: Danger/Dangerfile.swift;
  Line: 13;
-->

- :warning: Assigning default value from an optional variable may implicitly hide unintended `nil` values at development. Please make sure this is what you intended to do, and in that case please temporarily disable this rule at current line. (`optional_default_value`)